### ISSUE TITLE
`autoproj watch` gets frozen if the user needs to answer a configuration question

### ISSUE THREAD

**Issue by doudou** [rock-core/autoproj#200]
Ways to fix that:

- make `watch` fail in these cases, and wait for more changes. Should display a message that external tools can easily detect and show to the user (e.g. `ERROR: <msg>`)
- ?

**Comment by g-arjones** [comment]
Shouldn't autoproj skip interactive configuration if subcommand == watch?

**Comment by doudou** [comment]
Skipping the configuration is not really possible. The build setup depends on the configuration, so it can't be loaded or regen'ed.

This is why I was proposing to:
 - detect that interactive configuration is required
 - ignore the update event, and wait for more updates (i.e. user doing the configuration)

Unfortunately, the configuration will also be asked on e.g. update or build, which will have the same problem. Ultimately, we would have to be able to ask for the configuration value.

**Comment by g-arjones** [comment]
Yes, you are right. I think warning that the build is not properly configured and waiting for another event is the way to go.

**Comment by g-arjones** [comment]
Not an issue anymore, right? @doudou

**Comment by doudou** [comment]
Still an issue, however:
1. I agree we could probably use `--no-interactive` to use a temporary configuration until the user does configure (by e.g. running update or build explicitly)
2. `--no-interactive` only works for flags that have a default

I guess that doing (1) would still be better than nothing, but (2) will still bite every now and then
